component/service/src/main/java/org/exoplatform/social/service/malwareDetection/MalwareDetectionService.java

```diff
@@ -0,0 +1,89 @@
+  public static final String JCR_CONNECTOR = "jcr";
+  public static final String VALUES = "values";
+  
+  /**
+   * Add Malware detection Connector to the service
+   * @param malwareDetectionServiceConnector the Malware detection Connector to add
+   * @LevelAPI Experimental
+   */
+  public void addConnector (MalwareDetectionServiceConnector malwareDetectionServiceConnector) {
+    addConnector(malwareDetectionServiceConnector, false);
+  }
+  
+  /**
+   * Add malware detection Connector to the service
+   * @param malwareDetectionServiceConnector the malware detection connector to add
+   * @param override equal true if we can override an existing connector, false otherwise
+   * @LevelAPI Experimental
+   */
+  public void addConnector (MalwareDetectionServiceConnector malwareDetectionServiceConnector, Boolean override) {
+    if (connectors.containsKey(malwareDetectionServiceConnector.getType()) && override.equals(false)) {
+      LOGGER.error("Impossible to add connector {}. A connector with the same name has already been registered.", malwareDetectionServiceConnector.getType());
+    } else {
+      connectors.put(malwareDetectionServiceConnector.getType(), malwareDetectionServiceConnector);
+      LOGGER.info("A Malware detection Connector has been added: {}", malwareDetectionServiceConnector.getType());
+    }
+  }
+  
+  /**
+   * Gets all current connectors
+   * @return Connectors
+   * @LevelAPI Experimental
+   */
+  public Map<String, MalwareDetectionServiceConnector> getConnectors() {
+    return connectors;
+  }
+  
+  public void processInfectedItem(String infectedItemPath) {
+    if (infectedItemPath.contains(getJcrNodePathSeparator(infectedItemPath) + FILES_CONNECTOR + getJcrNodePathSeparator(infectedItemPath))) {
+      MalwareDetectionServiceConnector fileMalwareDetectionServiceConnector = (MalwareDetectionServiceConnector) getConnectors().get(FILES_CONNECTOR);
+      if(fileMalwareDetectionServiceConnector != null && fileMalwareDetectionServiceConnector.isEnable()) {
+        fileMalwareDetectionServiceConnector.sendInfectedItemNotification(infectedItemPath);
+        fileMalwareDetectionServiceConnector.cleanInfectedItem(infectedItemPath);
+      }
+    }
+    else if (infectedItemPath.contains(getJcrNodePathSeparator(infectedItemPath) + JCR_CONNECTOR + getJcrNodePathSeparator(infectedItemPath) + VALUES + getJcrNodePathSeparator(infectedItemPath))) {
+      MalwareDetectionServiceConnector jcrMalwareDetectionServiceConnector = (MalwareDetectionServiceConnector) getConnectors().get(JCR_CONNECTOR);
+      if(jcrMalwareDetectionServiceConnector != null && jcrMalwareDetectionServiceConnector.isEnable()) {
+        jcrMalwareDetectionServiceConnector.sendInfectedItemNotification(infectedItemPath);
+        jcrMalwareDetectionServiceConnector.cleanInfectedItem(infectedItemPath);
+      }
+    }
+  }
+  
+  public static String getJcrNodePathSeparator(String infectedFilePath) {
```

Comment:
```suggestion
  public static String getPathSeparator(String infectedFilePath) {
```
As this is not related to JCR we should not use 'JCR' in method name.